### ISSUE TITLE
Copy to local OS clipboard when SSH to remote servers

### ISSUE THREAD

**Issue by DravenJohnson** [gpakosz/.tmux#242]
> copy to OS clipboard (needs reattach-to-user-namespace on macOS, xsel or xclip on Linux)

Will this work when I use (*for example macOS), SSH into a remote server running Linux (*debian, ubuntu or archlinux) and use `tmux` with this dotfile? When I try to copy texts from remote server's tmux by using `C-b Enter` and `v -> y`, I can't get texts to my local machine.

Do i need both `reattach-to-user-namespace` and `xsel/xclip` install on both machines?

**Comment by gpakosz** [comment]
Hello @DravenJohnson 👋

There are different things.

On one side there's the OS clipboard, and the other side there are tmux buffers.

Setting `tmux_conf_copy_to_os_clipboard=true` in your `~/.tmux.conf.local` copy means that when you use bindings that normally save things into tmux buffers, they will also be copied into the OS clipboard. The data transfer is unidirectional, this configuration never reads **from** the OS clipboard.
This corresponds to the following sequence of actions, here with Vim bindings:
  - enter copy mode (`<prefix>` + `Enter`)
  - select some text (`v` then `h`, `j`, `k`, or `l`)
  - copy selection (`y`) ← with `tmux_conf_copy_to_os_clipboard=true` content goes both into tmux buffer and OS clipboard
  - after data entered OS clipboard, you paste it the usual way with e.g. `CTRL`+ `V` on Windows or `Command` + `V` on macOS

What I just described above requires `xsel` or `xclip` to be installed on Linux. For macOS, it depends: latest macOS versions don't need `reattach-to-user-namespace` anymore.

Now comes the SSH part. The `master` branch of my configuration doesn't address any data exchange between a remote and a local machine. But the work-in-progress `os-clipboard` branch does support this scenario:
  - SSH into a machine, where you have cloned this configuration and checked out the `os-clipboard` branch
  - launch tmux
  - enter copy mode (`<prefix>` + `Enter`)
  - select some text (`v` then `h`, `j`, `k`, or `l`)
  - copy selection (`y`) ← data is also copied back to the local machine through OSC 52 escape sequence

So yeah, you likely want to give the `os-clipboard` branch a try and report whether it works for you.

**Comment by DravenJohnson** [comment]
Great, thank you so much @gpakosz. It's nice you have really details answers. 

I'll try the `os-clipboard` and leave any feedback if there is any. Thank you.

**Comment by LubosRemplik** [comment]
Using `remote-clipboard` branch I can confirm it works with `ssh` client, it does not work when I use `mosh` client. Do you think `mosh` client blocks it somehow?

**Comment by gpakosz** [comment]
> Using `remote-clipboard` branch I can confirm it works with `ssh` client, it does not work when I use `mosh` client. Do you think `mosh` client blocks it somehow?

That branch is too old. I don't think `mosh` has anything to do with `node`?

**Comment by LubosRemplik** [comment]
This is likely related
https://github.com/mobile-shell/mosh/issues/967

Will try the patch mentioned in the issue and update here

**Comment by gpakosz** [comment]
I don't believe it's related. What you could do is finding out what creates these `nodes` processes. Maybe by using `pstree` to go back to the parent process

**Comment by LubosRemplik** [comment]
I am sorry, it is not `remote-clipboard` related, same behavior in `master` branch = `ssh` works, `mosh` does not work

**Comment by LubosRemplik** [comment]
@gpakosz sorry to confuse you, you are likely commenting different issue, let's go back to #403

**Comment by totoroyyb** [comment]
Wonder if there is any update or plan on this feature (for merging into the main)? It would be really helpful if I can yank text when sshed into the remote machine 😄 !
